### PR TITLE
Add a test for injecting an IRQ on a vCPU without a TCB associated

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
This PR adds a test for a scenario described in https://github.com/seL4/seL4/issues/1199.

Using the following configuration:
`../init-build.sh -DPLATFORM=odroidc4 -DSMP=1 -DARM_HYP=1`

You will currently see the kernel crash:
```
KERNEL DATA ABORT!
Faulting instruction: 0x8000012d38
FAR: 0x3a8 ESR (DFSR): 0x96000004
halting...
Kernel entry via Syscall, number: 1, Call
Cap type: 15, Invocation tag: 48
```

This test is ARM specific. I believe there are no other vCPU tests in sel4test right now so I added a new file.

### CODE REVIEW

File: apps/sel4test-tests/src/tests/vcpu.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#include "../helpers.h"
+
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+int test_vcpu_inject_without_tcb(env_t env)
+{
+  vka_object_t vcpu;
+  int error = vka_alloc_vcpu(&env->vka, &vcpu);
+
+  error = seL4_ARM_VCPU_InjectIRQ(vcpu.cptr, 0, 0, 0, 0);
+  // TODO: check for error
```
Comment (Ivan-Velickovic) [inline apps/sel4test-tests/src/tests/vcpu.c:23]: Finish

File: apps/sel4test-tests/src/tests/vcpu.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023
```
Comment (Ivan-Velickovic) [inline apps/sel4test-tests/src/tests/vcpu.c:2]:
```suggestion
 * Copyright 2024 UNSW
```

File: apps/sel4test-tests/src/tests/vcpu.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+/* This file contains tests related to vCPU syscalls. */
+
+#include <sel4/sel4.h>
+
+#include "../helpers.h"
+
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+int test_vcpu_inject_without_tcb(env_t env)
```
Comment (Ivan-Velickovic) [inline apps/sel4test-tests/src/tests/vcpu.c:14]: Should we explain why this (very basic) test exists? Or is that too much?

**Review by lsf37 — APPROVED** [review]
Thanks for the quick test case!

**Pull request merged**